packages/grid/x-data-grid-pro/src/utils/tree/removeDataRowFromTree.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  GRID_ROOT_GROUP_ID,
+  GridGroupNode,
+  GridRowId,
+  GridRowTreeConfig,
+  GridTreeNode,
+} from '@mui/x-data-grid';
+import { GridTreeDepths, GridRowTreeUpdatedGroups } from '@mui/x-data-grid/internals';
+import {
+  getNodePathInTree,
+  getGroupRowIdFromPath,
+  removeNodeFromTree,
+  updateGroupNodeIdAndAutoGenerated,
+  addActionToGroup,
+} from './utils';
+
+interface RemoveDataRowFromTreeParams {
+  /**
+   * ID of the data row to remove from the tree.
+   */
+  id: GridRowId;
+  /**
+   * Tree from which to remove the data row.
+   * This tree can be mutated but it's children should not.
+   * For instance:
+   *
+   * - `tree[nodeId] = newNode` => valid
+   * - `tree[nodeId].children.push(newNodeId)` => invalid
+   */
+  tree: GridRowTreeConfig;
+  /**
+   * Amount of nodes at each depth of the tree.
+   * This object can be mutated.
+   * For instance:
+   *
+   * - `treeDepths[nodeDepth] = treeDepth[nodeDepth] + 1` => valid
+   */
+  treeDepths: GridTreeDepths;
+  /**
+   * Object tracking the action performed on each group.
+   * Used to decide which groups to refresh on sorting, filtering, aggregation, ...
+   * If not defined, then the tracking will be skipped.
+   */
+  updatedGroups?: GridRowTreeUpdatedGroups;
```

Comment:
I assume like for the two others, `updatedGroups` receives an empty object and the method will fill it by using mutation

Reply:
As we discussed it, I will replace it will a small manager with an internal state to make the abstraction simpler from the outside